src/main/java/io/reactivex/internal/operators/observable/ObservableDematerialize.java

```diff
@@ -18,9 +18,9 @@
 import io.reactivex.internal.disposables.DisposableHelper;
 import io.reactivex.plugins.RxJavaPlugins;
 
-public final class ObservableDematerialize<T> extends ObservableSource<Try<Optional<T>>, T> {
+public final class ObservableDematerialize<T> extends ObservableSource {
```

Comment:
It's probably worth doing these as separate PRs. Rename/delete the existing `ObservableSource` and then do this rename. Otherwise it's likely to be extra confusing because of the name conflict.